### DISCUSSION TITLE
Find function on array column type

### DISCUSSION THREAD

**Discussion by LadIQe** [mikro-orm/mikro-orm#1652] [Q&A] [answered]
Hello, its possible to use `.find(<where>)` on column, which is type of array of numbers?

For example:

Entity
```
@Property({ default: [] })
category: number[]
```

Code:
`this.productRepository.find({ category: [2] }, { populate: ['price'] })`

Because right now, Im getting error
ERROR:  operator does not exist: text[] = integer at character 148
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
STATEMENT:  select "e0".*, "e1"."id" as "price_id" from "product" as "e0" left join "price" as "e1" on "e0"."uuid" = "e1"."product_uuid" where "e0"."category" in (2)

I know its working on column type of array of strings, but for numbers is somehow not working.

Using latest 4.5.2 version

Thank you

**Comment by LadIQe** [comment]
ok maybe I found problem

SQL for creating table: `"category" text[] not null default \'{}\'`

it looks like it ignores number type

**Reply by B4nan** [reply]
Try to set the column type explicitly to `columnType: 'int[]'` then, there is no way to infer the `number[]` via reflect metadata anyway (we only get `Array` from that here), so even if it is caused by `text[]`, you'd still have to provide the type.

**Reply by LadIQe** [reply]
yeah I tried this

`@Property({ type: new ArrayType(i => +i), columnType: 'integer[]', default: [] })`

got this
ERROR:  operator does not exist: integer[] = integer at character 148

maybe it doesnt like that array column

**Reply by B4nan** [reply]
try what I wrote in the other comment, the issue is your query, not in the column type. i believe the `$contains` operator will work here.

**Reply by LadIQe** [reply]
yeah it works with `$contains`, but I need to change type of my array (from number to string due to contains)

but hey it works at least :) thanks

**Comment by B4nan** [comment, marked as answer]
Maybe you could use `$contains` instead? (`@>` operator).

I don't think this is about the type, the problem will be in comparison of array to scalar.
